backend/infrahub/api/artifact.py

```diff
@@ -63,23 +64,23 @@ async def generate_artifact(
     branch_params: BranchParams = Depends(get_branch_params),
     account_session: AccountSession = Depends(get_current_user),
 ) -> None:
+    permission_manager = PermissionManager(account_session=account_session)
+    await permission_manager.load_permissions(db=db, branch=branch_params.branch)
+
     permission_decision = (
         PermissionDecisionFlag.ALLOW_DEFAULT
         if branch_params.branch.name in (GLOBAL_BRANCH_NAME, registry.default_branch)
         else PermissionDecisionFlag.ALLOW_OTHER
     )
-    for permission in [
+    permissions = [
         ObjectPermission(namespace="Core", name="Artifact", action=action.value, decision=permission_decision)
         for action in (PermissionAction.CREATE, PermissionAction.UPDATE)
-    ]:
-        has_permission = False
-        for permission_backend in registry.permission_backends:
-            if has_permission := await permission_backend.has_permission(
-                db=db, account_session=account_session, permission=permission, branch=branch_params.branch
-            ):
-                break
-        if not has_permission:
-            raise PermissionDeniedError(f"You do not have the following permission: {permission}")
+    ]
+
+    if not permission_manager.has_permissions(permissions=permissions):
+        raise PermissionDeniedError(
```

Comment:
it think it would be even cleaner if the permission_manager would raise the exception directly
this way it will guarantee that the error message will be generated consistently

Either we could do that within `has_permissions` directly or we can add another method that will check the permission and raise an exception